src/pages/CreateEventPage/CreateEventPage.tsx

```diff
@@ -0,0 +1,177 @@
+import React, { useState } from 'react';
+import { CreateEventWrapper, Form } from './styled';
+import { formatISO, format } from 'date-fns';
+import { createEvent } from '../../api/events';
+import TextField from '../../components/TextField';
+import DateField from '../../components/DateField';
+import TimeField from '../../components/TimeField';
+import SelectField from '../../components/SelectField';
+import TextAreaField from '../../components/TextAreaField';
+import Alert from '../../components/Alert';
+
+function CreateEvent() {
+  const [name, setName] = useState('');
+  const [image, setImage] = useState('');
+  const [imageAlt, setImageAlt] = useState('');
+  const [date, setDate] = useState(format(Date.now(), 'yyyy-MM-dd'));
+  const [starttime, setStarttime] = useState(format(Date.now(), 'HH:mm'));
+  const [endtime, setEndtime] = useState(format(Date.now(), 'HH:mm'));
+  const [description, setDescription] = useState('');
+  const [eventType, setEventType] = useState('');
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [email, setEmail] = useState('');
+  const [createWasSuccess, setCreateWasSuccess] = useState(false);
+  const [createFailed, setCreateFailed] = useState('');
+  const [clicked, setClicked] = useState(false);
+  const eventOptions: string[] = [
+    '',
+    'Subject matter event',
+    'Coding event',
+    'Project presentation',
+    'Workshop',
+  ];
+
+  function submitEvent() {
+    setCreateFailed('');
+    setCreateWasSuccess(false);
+    createEvent(
+      formatISO(Date.parse(date + ' ' + starttime)),
+      formatISO(Date.parse(date + ' ' + endtime)),
+      name,
+      description,
+      firstName,
+      lastName,
+      email
+    )
+      .then((result) => {
+        if (result) {
+          setCreateWasSuccess(true);
+        } else {
+          setCreateFailed('');
+        }
+      })
+      .catch((error) => {
+        setCreateFailed(error.message);
+      });
+    setClicked(false);
+  }
+
+  function isFormFilled(name: string, description: string, eventType: string) {
+    return name && description && eventType;
+  }
+
+  return (
+    <CreateEventWrapper>
+      {createFailed ? (
+        <Alert
+          heading="Failed to create event"
+          description={<p></p>}
```

Comment:
You can modify the `Alert` component to take an optional `description` instead of passing it an "empty" value 🙈.

Otherwise, it's also possible to throw something other than an `Error` in `createEvent`, e.g. an object containing an error code allowing you to distinguish between 400 and 500 HTTP responses.